common/src/main/scala/explore/components/AgGridHelpers.scala

```diff
@@ -0,0 +1,258 @@
+  }
+
+  // We return a View into the column
+  private def getter[A, B](lens: Lens[A, B]): js.Function1[ValueGetterParams, js.Any] =
+    parms => parms.data.asInstanceOf[View[A]].zoom(lens).asInstanceOf[js.Any]
+
+  // We use the View to set the values, not the grid functions.
+  // Return Boolean would prevent a re-render, but I don't think that's necessary if we
+  // use the isCancelAfterEnd method in the editing component to just cancel if the value
+  // is invalid.
+  private val noopSetter: js.Function1[ValueSetterParams, Boolean] = _ => true
+
+  // This receives a View[A] in the Props.
+  private def renderer[A](formatter: A => String) = ScalaComponent
+    .builder[View[A]]
+    .render_P((props: View[A]) => <.span(formatter(props.get)))
+    .build
+    .cmapCtorProps[ICellRendererParams](p => p.value.asInstanceOf[View[A]])
+    .toJsComponent
+    .raw
+
+  private val allowedKeys = List(KeyCode.Escape, KeyCode.Enter, KeyCode.Tab)
+
+  private def suppressKeys(parms: SuppressKeyboardEventParams): Boolean =
+    parms.editing && !allowedKeys.contains(parms.event.keyCode)
+
+  private def newId: NonEmptyString = NonEmptyString
+    .from("Id" + java.util.UUID.randomUUID().toString().replace("-", ""))
+    .getOrElse("idddd")
+
+  // TODO: Need to make a Backend for this to expose the ag-grid lifecycle methods
+  // (if that's how we expose them). Also probably want to edit State instead of view
+  // directly, as with the FormInputEV version.
+  private def makeFormInputEnumEditor[A](exclude: Set[A], includeCurrent: Boolean)(implicit
+    enum:                                         Enumerated[A],
+    display:                                      Display[A]
+  ) = {
+    val excludeFn: A => Set[A] = a => if (includeCurrent) exclude - a else exclude
+    ScalaComponent
+      .builder[View[A]]
+      .render_P(view => EnumViewSelect(id = newId, value = view, exclude = excludeFn(view.get)))
+      .build
+      .cmapCtorProps[ICellEditorParams](parms => parms.value.asInstanceOf[View[A]])
+      .toJsComponent
+      .raw
+  }
+
+  private def makeFormInputEVEditor[A](
+    validator:     ValidFormatInput[A],
+    changeAuditor: ChangeAuditor[A]
+  )(implicit eq:   Eq[A]) = {
+
+    @Lenses
+    case class State(original: A, current: A, editing: A)
+
+    class Backend($ : BackendScope[View[A], State])(implicit val eq: Eq[A]) {
+      val id = newId
+
+      // We need to add a little delay before looking for the input element. I think
+      // ag-grid is using some virtualization of it's own, and the input is not in
+      // the DOM immediately.
+      def setCursor(): Callback =
+        CallbackTo {
+          val inputOpt = Option(document.querySelector(s"#${id.value}").asInstanceOf[html.Input])
+          inputOpt.map { i =>
+            i.focus();
+            val len = i.value.length
+            i.setSelectionRange(0, len)
+          }
+        }.void.delayMs(50).completeWith(_ => Callback(()))
```

Comment:
`delayMs(0)` might work here. It will still queue the `Callback` instead of executing it immediately. Also you can do `.toCallback` on the resulting `AsyncCallback` (which does the same thing as the current code).

Reply:
Zero ms doesn't seem to work, but 1 ms does. Thanks for the `toCallback` info. I was trying to find something like that but missed it.